src/modules/azure.rs

```diff
@@ -194,6 +227,76 @@ mod tests {
         dir.close()
     }
 
+    #[test]
+    fn user_name_set_correctly() -> io::Result<()> {
```

Comment:
Please also add a test with a missing username, but correctly set-up subscription and the reverse.